src/Paprika/Merkle/MerkleNode.cs

```diff
@@ -0,0 +1,271 @@
+using System.Buffers.Binary;
+using System.Runtime.InteropServices;
+using Paprika.Crypto;
+using Paprika.Data;
+
+namespace Paprika.Merkle;
+
+public static class Node
+{
+    public enum Type : byte
+    {
+        Leaf,
+        Extension,
+        Branch,
+    }
+
+    public static ReadOnlySpan<byte> ReadFrom(ReadOnlySpan<byte> source, out Type nodeType, out Leaf leaf, out Extension extension, out Branch branch)
```

Comment:
Should this method return anything at all? Maybe `void` or maybe the `nodeType` could be actually returned here?